rectools/model_selection/cross_validate.py

```diff
@@ -0,0 +1,139 @@
+import typing as tp
+
+import pandas as pd
+
+from rectools.columns import Columns
+from rectools.dataset import Dataset, Features, IdMap, Interactions
+from rectools.metrics import calc_metrics
+from rectools.metrics.base import MetricAtK
+from rectools.models.base import ModelBase
+from rectools.types import ExternalIds
+
+from .splitter import Splitter
+
+
+def gen_2x_internal_ids_dataset(
+    interactions_internal_df: pd.DataFrame,
+    user_features: tp.Optional[Features],
+    item_features: tp.Optional[Features],
+) -> Dataset:
+    """
+    Make new dataset based on given interactions and features from base dataset.
```

Comment:
I suggest to rename this func and provide reasons for usage in docstring (see other comment below)